x/ibc/04-channel/types/packet.go

```diff
@@ -0,0 +1,108 @@
+package types
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/ibc/04-channel/exported"
+	host "github.com/cosmos/cosmos-sdk/x/ibc/24-host"
+)
+
+var _ exported.PacketI = Packet{}
```

Comment:
Why are we differentiate `PacketI` and `Packet`?

Reply:
`PacketI` is the interface whereas `Packet` is the concrete type. Are you suggesting we should have one or the other?

Reply:
- `Sequence, SourcePort, DestinationPort, SourceChannel, DestinationChannel` information is embedded inside the keypath, so we should make it a field of the message type instead of Packet.
- Therefore the data actually being stored inside the state should be a struct type of `(timeout, data)` where the `data` is marshaled bytes passed from the application logic.